lib/Config/LocationAwareConfigRepository.php

```diff
@@ -95,7 +107,9 @@ public function __construct(
         ?string $writeTargetEnvVariableName,
         ?string $defaultWriteLocation = null,
         ?string $legacyConfigFile = null,
-        mixed $loadLegacyConfigCallback = null
+        mixed $loadLegacyConfigCallback = null,
+        ?string $writeTarget = null,
+        ?array $options = null
```

Comment:
This is a BC break, we need to use the parameter `$storageDirectory` instead. 

We pass the config array in `$storageDirectory` and then check whether it's an `array` (new config) or a `string` (old config). 

Additionally we deprecate both `$writeTargetEnvVariableName` and   `$defaultWriteLocation`.